### PR TITLE
scrollToOptions dictionary feature detection

### PR DESCRIPTION
**Pull request by asmarcz**
Fixes https://github.com/Modernizr/Modernizr/issues/2433

### CODE REVIEW

**Comment by rejas** [comment]
hi @asmarcz sorry for the long pause. could you rebase your PR against the latest master so that we can see if the tests dont fail anymore?

File: feature-detects/scroll-to-options.js [hunk 1]
```diff
@@ -0,0 +1,27 @@
+/*!
+{
+  "name": "scrollToOptions dictionary",
+  "property": "scrolltooptions"
+}
```
Comment (rejas) [inline feature-detects/scroll-to-options.js]: coudl you add at least the tags for author (yourself :-) to this? also if there is a caniuse property check for this add this too as well as other links that might be useful.

File: lib/config-all.json [hunk 2]
```diff
@@ -253,6 +253,7 @@
     "requestanimationframe",
     "script/async",
     "script/defer",
+    "scroll-to-options",
```
Comment (rejas) [inline lib/config-all.json]: could you move the test into the css directory? I think its more fitting there

Comment (asmarcz) [reply lib/config-all.json]: @rejas  Why would you move it into the css directory? It is a test of JS function and its behaviour. Sorry for the pause, didn't read GitHub notifications.

File: feature-detects/scroll-to-options.js [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+!*/
+define(['Modernizr'], function (Modernizr) {
```
Comment (rejas) [inline feature-detects/scroll-to-options.js]: Please use helper classes like this:

define(['Modernizr', 'createElement', 'getBody'], function (Modernizr, createElement, getBody) {

File: feature-detects/scroll-to-options.js [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+  Modernizr.addTest('scrolltooptions', function () {
+    var returnTo = window.pageYOffset;
+    var needsFill = document.body.clientHeight <= window.innerHeight;
```
Comment (rejas) [inline feature-detects/scroll-to-options.js]: so now use
    var body = getBody();
    var needsFill = body.clientHeight <= window.innerHeight;

(and for the other occurences of the body element too

File: feature-detects/scroll-to-options.js [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+    if (needsFill) {
+      var div = document.createElement('div');
```
Comment (rejas) [inline feature-detects/scroll-to-options.js]: same here, use helper class 
      var div = createElement('div');

File: lib/config-all.json (hunk 2 above)
Comment (rejas) [reply lib/config-all.json]: you're right, forget that :-)

File: feature-detects/scroll-to-options.js [hunk 6]
```diff
@@ -0,0 +1,27 @@
+/*!
+{
+  "name": "scrollToOptions dictionary",
+  "property": "scrolltooptions"
+}
+!*/
```
Comment (rejas) [inline feature-detects/scroll-to-options.js]: some docs on what the test does and expects would be very helpful

**Comment by rejas** [comment]
@asmarcz still working on the requested changes? please also make sure that the test dont fail

**Comment by asmarcz** [comment]
No, sorry. I don't plan to waste another time with this time scattered issue.

**Pull request closed without merging**